### PR TITLE
Refactoring FindMeetingQuery

### PR DESCRIPTION
**Pull request by sarahz916**
Tried to clean up code in FindMeetingQuery.java.

### CODE REVIEW

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 1]
```diff
@@ -63,52 +70,40 @@
...
     }
     return solutions;
-    //cannot conflict with the collection of events given.
-    //make sure to check end point of time range is inclusive. 
+
   }
 
-  /**Splits a TimeRange into two time or shotens one TimeRange due to conflict with a given event */
+  /**Splits a TimeRange into two TimeRanges if conflicting event is contained in TimeRange
+   or shotens one TimeRange due to conflict with a given event */
```
Comment (JKubs15) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: shortens

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 2]
```diff
@@ -63,52 +70,40 @@
...
-              else{
-                  TimeRange shortenedSlot = TimeRange.fromStartEnd(slot.start(), event.getWhen().start(), false);
-                  if (checkTimeRangeisLongEnough(duration, shortenedSlot)){
-                      solutions.add(shortenedSlot);
-                  }
-              }
-              
+          
+          else if (slot.overlaps(event.getWhen())){ 
+              // Need to take out slot and event overlap.
```
Comment (JKubs15) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: indentation

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 3]
```diff
@@ -63,52 +70,40 @@
...
+            ArrayList<TimeRange> nonConflictingTimes = new ArrayList<>();
+            nonConflictingTimes.addAll(takeOutConflict(slot, event));
```
Comment (JKubs15) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: You changed a similar circumstance to happen in one line, is there a reason this should be different?

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 4]
```diff
@@ -63,52 +70,40 @@
...
+            System.out.println(nonConflictingTimes);
+            // Check if nonconflicting times are long enough for meeting request duration
+            for (TimeRange nonconflictingTime: nonConflictingTimes){
+                if (checkTimeRangeisLongEnough(duration, nonconflictingTime)){
```
Comment (JKubs15) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: I'd suggest isTimeRangeLongEnough so the bool expectation is clear.

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 5]
```diff
@@ -63,52 +70,40 @@
...
-              }
-              
+          
+          else if (slot.overlaps(event.getWhen())){ 
+              // Need to take out slot and event overlap.
+            ArrayList<TimeRange> nonConflictingTimes = new ArrayList<>();
+            nonConflictingTimes.addAll(takeOutConflict(slot, event));
+            System.out.println(nonConflictingTimes);
+            // Check if nonconflicting times are long enough for meeting request duration
+            for (TimeRange nonconflictingTime: nonConflictingTimes){
```
Comment (JKubs15) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: Capitalization is different between the list and the element.

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 6]
```diff
@@ -125,9 +120,46 @@ private boolean overlappingAttendees(MeetingRequest request, Event event){
...
+  conflict with event. */
+  private TimeRange shortenTimeRange(TimeRange slot, Event event){
+    if (event.getWhen().end() > slot.start()){
+    TimeRange shortenedSlot = TimeRange.fromStartEnd(event.getWhen().end(), slot.end(), false);
+    return shortenedSlot;
+    }
+    else{
+    TimeRange shortenedSlot = TimeRange.fromStartEnd(slot.start(), event.getWhen().start(), false);
+    return shortenedSlot;
+    }
```
Comment (JKubs15) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: Indentation?

**Review by JKubs15 — COMMENTED** [review]
I know it's a bit late, but generally we use different capitalization standards for functions and variables (maybe Java is diffferent). In C++, we use camel for functions and snake for variables, doesn't need to change here, but a nice subtle variation that makes each easily identifiable.

**Comment by query** [comment]
> I know it's a bit late, but generally we use different capitalization standards for functions and variables (maybe Java is diffferent). In C++, we use camel for functions and snake for variables, doesn't need to change here, but a nice subtle variation that makes each easily identifiable.

Java is in fact different; functions and variables are pretty much universally lowerCamelCase.

**Review by JKubs15 — APPROVED** [review]

**Pull request merged**